substrate/frame/support/test/tests/pallet_ui/pass/no_warning.rs

```diff
@@ -0,0 +1,37 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[deny(warnings)]
```

Comment:
We should already be forbidding warnings here:  

https://github.com/paritytech/polkadot-sdk/blob/bc53b9a03a742f8b658806a01a7bf853cb9a86cd/substrate/frame/support/test/tests/pallet_ui.rs#L30-L31

But it could still help to have this test here (without the `deny` tag i mean).

Reply:
Don't we set this already in CI?

Reply:
Yea also, but i think this forks the compiler without copying the env.

Reply:
indeed, I changed it to a simple pass test with a storage.

anyway the warning only happen on nightly for the moment